interp/scope.go

```diff
@@ -135,9 +135,25 @@ func (s *scope) rangeChanType(n *node) *itype {
 			return t
 		}
 	}
-	if c := n.child[1]; c.typ != nil && c.typ.cat == chanT {
+
+	c := n.child[1]
+	if c.typ != nil && c.typ.cat == chanT {
 		return c.typ
 	}
+	if isBinCall(c) &&
```

Comment:
It would be better here to check for `c.typ.cat` of `valueT` and a `c.typ.rtype.Kind()` of `reflect.Chan`. Also you will need to hoist the `c.typ == nil`.

Reply:
Will do, thanks.

Reply:
(although I'm not super clear about how my previous code was actually working :sweat_smile: )